src/VisualStudio/Core/Def/Implementation/ProjectSystem/AbstractProject_Options.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Host;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem
+{
+    internal abstract partial class AbstractProject
+    {
+        private CommandLineArguments _lastParsedCommandLineArguments;
+
+        /// <summary>
+        /// Creates and sets new options using the last parsed command line arguments.
+        /// </summary>
+        protected void UpdateOptions()
+        {
+            Contract.ThrowIfNull(_lastParsedCommandLineArguments);
+
+            var newParseOptions = CreateParseOptions(_lastParsedCommandLineArguments);
+            var newCompilationOptions = CreateCompilationOptions(_lastParsedCommandLineArguments, newParseOptions);
+            if (newCompilationOptions == CurrentCompilationOptions && newParseOptions == CurrentParseOptions)
+            {
+                return;
+            }
+
+            SetOptions(newCompilationOptions, newParseOptions);
+        }
+
+        /// <summary>
+        /// Creates and sets new options using the given parsed command line arguments.
+        /// </summary>
+        protected void SetArgumentsAndUpdateOptions(CommandLineArguments commandLineArguments)
+        {
+            _lastParsedCommandLineArguments = commandLineArguments;
+            UpdateOptions();            
+        }
+
+        protected void SetOptions(CompilationOptions newCompilationOptions, ParseOptions newParseOptions)
```

Comment:
This is still used by VisualBasicProject (legacy VB shim) [here](https://github.com/dotnet/roslyn/pull/12440/files/7f3e7c8d7f79d6e14d6d9c649d2eef9db130495a#diff-dd53e8d47425333c43f1961e0a9df52bR493) - Same as before this change [here](http://source.roslyn.io/#Microsoft.VisualStudio.LanguageServices.VisualBasic/ProjectSystemShim/VisualBasicProject.vb,484)